packages/headless/src/features/search-parameters/search-parameter-serializer.ts

```diff
@@ -1,38 +1,98 @@
 import {SearchParameters} from './search-parameter-actions';
 
+const delimiter = '&';
+const equal = '=';
+
 export function buildSearchParameterSerializer() {
   return {serialize, deserialize};
 }
 
 function serialize(obj: SearchParameters) {
-  const fragment = Object.entries(obj)
-    .map(([key, val]) => `${key}=${val}`)
-    .join('&');
+  return Object.entries(obj)
+    .map(serializePair)
+    .filter((str) => str)
+    .join(delimiter);
+}
+
+function serializePair(pair: [string, unknown]) {
+  const [key, val] = pair;
+
+  if (!isValidKey(key)) {
+    return '';
+  }
+
+  if (key === 'f') {
+    return isFacetObject(val) ? serializeFacets(key, val) : '';
+  }
+
+  return `${key}${equal}${val}`;
+}
 
-  return fragment;
+function isFacetObject(obj: unknown): obj is Record<string, string[]> {
+  if (obj && typeof obj === 'object') {
+    const invalidEntries = Object.entries(obj).filter(([key, value]) => {
```

Comment:
I'm getting a few console errors here:
```
src/features/search-parameters/search-parameter-serializer.ts:33:43 - error TS2769: No overload matches this call.
[0]   Overload 1 of 2, '(o: { [s: string]: unknown; } | ArrayLike<unknown>): [string, unknown][]', gave the following error.
[0]     Argument of type 'object | null' is not assignable to parameter of type '{ [s: string]: unknown; } | ArrayLike<unknown>'.
[0]       Type 'null' is not assignable to type '{ [s: string]: unknown; } | ArrayLike<unknown>'.
[0]   Overload 2 of 2, '(o: {}): [string, any][]', gave the following error.
[0]     Argument of type 'object | null' is not assignable to parameter of type '{}'.
[0]       Type 'null' is not assignable to type '{}'.
[0]
[0] 33     const invalidEntries = Object.entries(obj).filter(([key, value]) => {
[0]                                              ~~~
[0]
[0]
[0] src/features/search-parameters/search-parameter-serializer.ts:33:57 - error TS7031: Binding element 'key' implicitly has an 'any' type.
[0]
[0] 33     const invalidEntries = Object.entries(obj).filter(([key, value]) => {
[0]                                                            ~~~
[0]
[0] src/features/search-parameters/search-parameter-serializer.ts:33:62 - error TS7031: Binding element 'value' implicitly has an 'any' type.
[0]
[0] 33     const invalidEntries = Object.entries(obj).filter(([key, value]) => {
[0]                                                                 ~~~~~
```
Casting obj to an object removes them for me: `Object.entries(obj as object).fil...`

Reply:
Hmm, could it be a local typescript issue? I do not see this error locally, or on the CI build.